### DISCUSSION TITLE
Timeouts for states

### DISCUSSION THREAD

**Discussion by hasys** [serverlessworkflow/specification#726] [General] [answered]
Hello,

It looks like there is a typo in section [States Timeout Definition](https://github.com/serverlessworkflow/specification/blob/main/specification.md#states-timeout-definition). At least there is a contradiction between different states declaration (at the same document) and this States timeout definition info.

For an example [Inject](https://github.com/serverlessworkflow/specification/blob/main/specification.md#Inject-State) and [Sleep](https://github.com/serverlessworkflow/specification/blob/main/specification.md#Sleep-State) states there is missing information of support timeout, but in States Timeout Definition it says that ALL states support timeout. Is timeout parameter missing in states description. Maybe timeout is inheriting the value from the workflow, or Sleep and Inject states just should be excluded from ALL states support timeout? Like: "All states except Sleep and Inject are support timeout.".

Thank you for your clarification!

**Comment by cdavernas** [comment, marked as answer]
@hasys inject and sleep cannot, by definition, support timeout. 
The inject state should either complete or fail, and the sleep state should imho be gone, as the same can be achieved with actions waitBefore/waitAfter.
